File: user-service/src/main/java/io/ecommerce/userservice/common/utils/PageUtils.java
package io.ecommerce.userservice.common.utils;

import org.springframework.data.domain.Pageable;

/**
 * @author : choi-ys
 * @date : 2022/01/08 5:55 오후
 */
public class PageUtils {

    public static Pageable pageNumberToIndex(Pageable pageable) {
        return pageable.getPageNumber() != 0 ?
                pageable.withPage(pageable.getPageNumber() - 1) : pageable;
    }
}

File: order-service/src/main/kotlin/io/ecommerce/orderservice/common/utils/PageUtils.kt
package io.ecommerce.orderservice.common.utils

import org.springframework.data.domain.Pageable

/**
 * @author : choi-ys
 * @date : 2022/01/09 6:14 오후
 */
class PageUtils {
    companion object {
        fun pageNumberToIndex(pageable: Pageable): Pageable {
            return if (pageable.pageNumber != 0)
                pageable.withPage(pageable.pageNumber - 1)
            else pageable
        }
    }
}